src/utils/Ponyfills.js

```diff
@@ -0,0 +1,15 @@
+// This file is intented to have multiple exports
+// eslint-disable-next-line import/prefer-default-export
+export function convertUint8ArrayToString(array) {
+    if (window.TextDecoder !== undefined) {
+        return new TextDecoder().decode(array);
+    }
+
+    // Avoid the String.fromCharCode.apply(null, array) shortcut, which
+    // throws a "maximum call stack size exceeded" error for large arrays.
```

Comment:
This could at least process the array in chunks, rather than one element at a time.

Maybe it's not worth the added complexity though, if that approach is fast enough.